docs/oid-info.md

```diff
@@ -58,11 +60,89 @@ the git ref that the workflow run was based upon.
 This specifies the username identity in the OtherName Subject Alternative Name, as
 defined by [RFC5280 4.2.1.6](https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.6).
 
+
+### 1.3.6.1.4.1.57264.1.8 | Source URI
+
+Should include a fully qualified source repository URL that the build was based
+on.
+
+### 1.3.6.1.4.1.57264.1.9 | Source Digest
+
+An immutable reference to a specific version of the source code that the build
+was based upon. Should include the digest type followed by the digest, e.g.
+`sha1:abc123`.
+
+### 1.3.6.1.4.1.57264.1.10 | Source Ref
+
+The source ref that the build run was based upon. For example: `refs/head/main`.
+
+### 1.3.6.1.4.1.57264.1.11 | Source Identifier
+
+The stable identifier for the source repository the workflow was based upon.
+
+### 1.3.6.1.4.1.57264.1.12 | Source Owner URI
+
+Fully qualified URL for the owner of the source repository that the build was based
+on.
+
+### 1.3.6.1.4.1.57264.1.13 | Source Owner Identifier
+
+The stable identifier for the owner of the source repository that the workflow was based upon.
+
+### 1.3.6.1.4.1.57264.1.14 | Build Config URI
+
+A reference to the initiating build instructions.
+
+### 1.3.6.1.4.1.57264.1.15 | Build Config Digest
+
+An immutable reference to the specific version of the top-level build
+instructions. Should include the digest type followed by the digest, e.g.
+`sha1:abc123`.
+
+### 1.3.6.1.4.1.57264.1.16 | Build Signer URI
+
+Reference to specific build instructions that are responsible for signing. Can be the same as Build Config URI. For example a reusable workflow in GitHub Actions or a Circle CI Orbs.
+
+Note this value would also be set in the Subject Alternative Name (SAN).
+
+### 1.3.6.1.4.1.57264.1.17 | Build Signer Digest
+
+An immutable reference to the specific version of the build instructions that is responsible for signing. Should include the digest type followed by the digest, e.g. `sha1:abc123`.
+
+### 1.3.6.1.4.1.57264.1.18 | Build Trigger
+
+The event or action that triggered the build.
+
+### 1.3.6.1.4.1.57264.1.19 | Runner Environment
+
+For platforms to specify whether the build took place in platform-hosted cloud infrastructure or customer-hosted infrastructure. For example: `platform-hosted` and `self-hosted`.
+
```

Comment:
```suggestion

### 1.3.6.1.4.1.57264.1.20 | Runner Execution Identifier

An immutable value that can uniquely identify the build' sexecution. For example: `4043059413`

```

Reply:
> but in the meantime being able to provide a non-falsifiable audit trail back to the specific build execution that requested the cert feels like an interesting/important property – especially in forensic contexts.

Would this one that is deprecated eventually? 

I understand the benefit of one, but this seems like a temporary solution to the broader problem of actually creating a **non-falsifiable builder** to get that audit trail

Reply:
I think it actually makes sense. I would think of this kind of similar to the serial number for a cert. It adds a trust relation to it's origin (i.e its not at all intended for any policy decision).

Today, if we would like to understand if certificate from Fulcio is fraudulent or not, we have very limited information to go on. We have the the time it was created, but that's about it, the rest of the correlation would be a bit fuzzy. If we have the e.g. the run id, we know it's origin and can easier compare logs.

Having transparency is of course what enables this, similar to how it's done with WebPKI. What I think can be a problem here, is that the order of certificates created is on a different scale for signing certificates compared to domain certificates, so a "weak mapping" is harder to implement, and so including the run id would create a stronger link that's easier to automate on.

Reply:
> I would think of this kind of similar to the serial number for a cert

Yeah, I was thinking about this, and wondering if it's something like a "session id" for the dex/email case.

My one question is: does it uniquely identify the identity/trust of the run? From what I understand, a single run_id can reference maybe many reusable workflows, which have different "identities". (Or is it the combo of the run_id and the job_workflow_ref/sha that can give you a unique reference and auditable issuance to the signer?)

What do you think then is the criticality of this extension? Honestly now maybe I can't decide if it's a MUST or very optional. MUST because it's an audit number to the OIDC token (does any other field suffice?), or optional because it's not related really to a useful part of the "identity" of the signer. Maybe I don't understand this: are you trying to audit the OIDC token issuance, or the invocation it was used in? The latter is definitely not "identity" policy related, so would be optional.

Reply:
Yes, you are correct that we need to figure out how to make it unique, like a combination of`run_id/number/attempt`. This information together with e.g. the workflow_ref or sha as you propose. Not sure if the extension need to be cater for this, or if the "backing system" should be responsible for putting them altogether, I think that may be more useful as it's a bit more flexible/future-proof?

I would say it's RECOMMENDED? As from a general signer/identity perspective, it doesn't really add much?

Reply:
A potential use-case I can see is for a CI provider to monitor the CT log and correlate each certificate with a workflow run to scan for forged certificates.

Reply:
> I would say it's RECOMMENDED? As from a general signer/identity perspective, it doesn't really add much?

I'd be happy with that! It actually works for almost all extensions as well "here's the description of the extension and here's the "usage hint"" (which can help distinguish between critical for identity verification and informational and for what). WDYT @haydentherapper? 

Ideally the critical for identity verification line up with the required extensions Fulcio provides.

Other things, we should document and be clear what it's purpose is -- so that if some client abuses usage, it's up to them really but at least the intent for the extension is clear.

Reply:
I like this addition! As mentioned, this needs to include the run id and attempt to make it unique in GHA. Thoughts on following the lead of SLSA 1.0 draft and constructing a URI for this? https://github.com/slsa-framework/slsa/blob/main/docs/github-actions-workflow/v0.1/index.md#metadata

Something like `Run Invocation URI: github.server_url + "/actions/runs/" + github.run_id + "/attempts/" + github.run_attempt`. This way the value is immediately parseable and useful without having to know some custom format for GHA, e.g. `github.run_id + '-' + github.run_attempt`.

Reply:
`Build Invocation URI`?

Reply:
While I don't think strict alignment with SLSA is relevant for Fulcio, I do think it's valuable to rely on prior art, so I like the idea or reusing this format.

Reply:
Added this as `Run Invocation URI` in the current proposal.